### ISSUE TITLE
Comment highlighting in OCamlyacc

### ISSUE THREAD

**Issue by adamjstewart** [sublimehq/Packages#1364]
TL;DR The OCamlyacc syntax highlighting doesn't support Menhir comments.

OCaml comes with a builtin parser generator called `ocamlyacc`, which reads a grammar definition (`parser.mly`) and generates a parser. Grammar definitions have the [following format](https://caml.inria.fr/pub/docs/manual-ocaml/lexyacc.html#sec324):
```ocaml
%{
  header
%}
  declarations
%%
  rules
%%
  trailer
```
> Comments are enclosed between `/*` and `*/` (as in C) in the "declarations" and "rules" sections, and between `(*` and `*)` (as in OCaml) in the "header" and "trailer" sections.

Sublime's OCamlyacc syntax highlighting matches this comment style perfectly. In fact, a little too perfectly.

Menhir is a drop-in replacement for Ocamlyacc, which supports mostly the same syntax but has many additional features. I can't speak for the rest of the community, but from what I've seen Menhir has almost entirely replaced usage of Ocamlyacc.

Menhir supports a broader array of comment options (see section 3 of the [Menhir manual](http://gallium.inria.fr/~fpottier/menhir/manual.pdf)):

> Comments are C-style (surrounded with `/*` and `*/`, cannot be nested), C++-style (announced by `//` and extending until the end of the line), or OCaml-style (surrounded with `(*` and `*)`, can be nested). Of course, inside OCaml code, only OCaml-style comments are allowed.

I would like to propose that we extend the OCamlyacc syntax highlighting to support Menhir comments as well. The "header" and "trailer" sections will still only support OCaml-style comments, but the "declarations" and "rules" sections will support OCaml-style, C-style, and C++-style comments.

[This file](https://github.com/sublimehq/Packages/blob/master/OCaml/OCamlyacc.sublime-syntax) appears to be the one that needs to be changed, but I don't know anything about how to modify or test this file.
